src/main/java/com/google/cloud/tools/gradle/appengine/core/CloudSdkDownloader.java

```diff
@@ -17,11 +17,199 @@
 
 package com.google.cloud.tools.gradle.appengine.core;
 
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkNotFoundException;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkOutOfDateException;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkVersionFileException;
+import com.google.cloud.tools.managedcloudsdk.BadCloudSdkVersionException;
+import com.google.cloud.tools.managedcloudsdk.ConsoleListener;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVersionMismatchException;
+import com.google.cloud.tools.managedcloudsdk.ProgressListener;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.Version;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExecutionException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandExitException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponentInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstallerException;
+import com.google.cloud.tools.managedcloudsdk.update.SdkUpdater;
 import java.io.File;
+import java.io.IOException;
+import org.gradle.api.GradleException;
+import org.gradle.api.logging.Logger;
 
-interface CloudSdkDownloader {
+public class CloudSdkDownloader {
 
-  File downloadSdk(String version);
+  private Logger logger;
 
-  boolean isSdkValid(String version, File home);
+  private SdkInstaller installer;
```

Comment:
We should be able to mock these out without creating these member variables. If we inject ManagedCloudSdk into this class, we can just mock the return values for the methods newInstaller, newUpdater and so on to return mocks.

Reply:
I just realized how many times I said mock in this comment.

Basically, find a way to pass in the ManagedCloudSdk at construction time. And in our tests, just mock the internals out.